SingularityService/src/main/java/com/hubspot/singularity/data/RequestManager.java

```diff
@@ -133,20 +133,22 @@ private String getPendingPath(String requestId, String deployId) {
     return ZKPaths.makePath(PENDING_PATH_ROOT, new SingularityDeployKey(requestId, deployId).getId());
   }
 
-  private String getPendingPath(SingularityPendingRequest pendingRequest) {
-    String nodeName = pendingQueueKey(pendingRequest);
+  private String getPendingPath(SingularityPendingRequest pendingRequest, boolean forceImmediate) {
+    String nodeName = pendingQueueKey(pendingRequest, forceImmediate);
     return ZKPaths.makePath(PENDING_PATH_ROOT, nodeName);
   }
 
-  private String pendingQueueKey(SingularityPendingRequest pendingRequest) {
+  private String pendingQueueKey(SingularityPendingRequest pendingRequest, boolean forceImmediate) {
     SingularityDeployKey deployKey = new SingularityDeployKey(pendingRequest.getRequestId(), pendingRequest.getDeployId());
     if (pendingRequest.getPendingType() == PendingType.ONEOFF) {
       return String.format("%s%s", deployKey, pendingRequest.getTimestamp());
     } else if (pendingRequest.getPendingType() == PendingType.IMMEDIATE) {
       Optional<SingularityPendingRequest> existingRequest = getPendingRequest(pendingRequest.getRequestId(), pendingRequest.getDeployId());
-      if (existingRequest.isPresent() &&
-          (existingRequest.get().getPendingType() == PendingType.TASK_DONE
-          || existingRequest.get().getPendingType() == PendingType.NEW_DEPLOY)) {
+      boolean markImmediate = forceImmediate
+          || (existingRequest.isPresent()
```

Comment:
I think this may still be causing us some issues. I've seen a scheduled request or two pop through that never got a pending task in place after a successful deploy, I think due to the two possible deletes and the fact that they could be enqueued between those attempts. We also have a weird bit going on with the `getPendingPath(String requestId, String deployId)` implementation which returns a different version of the path when we call `getPendingRequest`

I'm wondering about the consequences of returning `String.format("%s%s", deployKey, pendingRequest.getTimestamp())` for all cases. Since we sort the pending requests by timestamp, we will still be able to put multiple in the queue. The work then falls to the scheduler to de-dupe and process appropriately. We should check out where else we might be relying on the `EXISTED` response to see if this is possible